wbooks/src/BookDetail.js

```diff
@@ -1,51 +1,89 @@
 import React, { Component } from 'react';
-import myBooks from './books.json';
 import { Link } from 'react-router-dom';
 import defaultBookIcon from './default_book.svg';
 import './BookDetail.css';
+import BooksApi from './services/BooksApi'
+import BookCover from './BookCover'
 
 class BookDetail extends Component {
+  constructor() {
+    super();
+    this.state = {
+      book: [],
+      notFound: false,
+      fetchingBooks: false}
+  };
+
+  componentDidMount() {
+    this.setState({fetchingBook: true})
+    BooksApi.getBook(this.props.match.params.id).then((response) => {
+      this.setBook(response.data);
+    }).catch((error) => {
+      if (error.response.status === '400') {
+        this.setState({notFound: true, fetchingBook: false});
+      }
+    })
+  }
+
+  setBook = (book) => {
+    this.setState({book, fetchingBook: false})
+  }
+
   render() {
-    const book = myBooks.find(book => book.id === this.props.match.params.id);
+    if (this.state.fetchingBook) {
+      return (
+        <div className="body">
+          <b className='loading'>Se está buscando el libro</b>
+        </div>
+      )
+    } else if (this.state.notFound) {
```

Comment:
You don't need the else if you use return above.
just `if (this.state.notFound) {`